src/OpenLoco/src/Vehicles/Vehicle1.cpp

```diff
@@ -126,13 +130,167 @@ namespace OpenLoco::Vehicles
         return true;
     }
 
+    // 0x004F8974
+    // TODO: Move to track data
+    std::array<uint16_t, 44> _trackIdToCurveSpeedFraction = {
+        0xFFFF, // 1.00
+        0xFFFF, // 1.00
+        0x0CCD, // 0.05
+        0x0CCD, // 0.05
+        0x199A, // 0.10
+        0x199A, // 0.10
+        0x2666, // 0.15
+        0x2666, // 0.15
+        0x4000, // 0.25
+        0x4000, // 0.25
+        0x4000, // 0.25
+        0x4000, // 0.25
+        0x2666, // 0.15
+        0x2666, // 0.15
+        0xFFFF, // 1.00
+        0xFFFF, // 1.00
+        0xFFFF, // 1.00
+        0xFFFF, // 1.00
+        0x199A, // 0.10
+        0x199A, // 0.10
+        0x199A, // 0.10
+        0x199A, // 0.10
+        0x199A, // 0.10
+        0x199A, // 0.10
+        0x199A, // 0.10
+        0x199A, // 0.10
+        0xFFFF, // 1.00
+        0xFFFF, // 1.00
+        0x0CCD, // 0.05
+        0x0CCD, // 0.05
+        0x0CCD, // 0.05
+        0x0CCD, // 0.05
+        0x0CCD, // 0.05
+        0x0CCD, // 0.05
+        0xFFFF, // 1.00
+        0xFFFF, // 1.00
+        0xFFFF, // 1.00
+        0xFFFF, // 1.00
+        0x0CCD, // 0.05
+        0x0CCD, // 0.05
+        0x0CCD, // 0.05
+        0x0CCD, // 0.05
+        0x0CCD, // 0.05
+        0x0CCD, // 0.05
+    };
+
+    // 0x004B98DA
+    static void railProduceCrossingWhistle(const Vehicle2& veh2)
+    {
+        Vehicle train{ veh2.head };
+        auto* vehObj = ObjectManager::get<VehicleObject>(train.cars.firstCar.front->objectId);
+
+        if (vehObj->numStartSounds == 0)
+        {
+            return;
+        }
+
+        gPrng1().randNext(); // ???
```

Comment:
remove?

Reply:
Sadly if i want deterministic with vanilla simulation i need to keep this pointless Rand. I guess i could add a todo delete